File: src/main/java/kitchenpos/menus/tobe/dto/CreateMenuRequest.java
package kitchenpos.menus.tobe.dto;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.UUID;

public class CreateMenuRequest {
    private String name;
    private BigDecimal price;
    private UUID menuGroupId;
    private boolean displayed;
    private List<MenuProductRequest> menuProducts;

    public CreateMenuRequest() {}

    public CreateMenuRequest(final String name, final BigDecimal price, final UUID menuGroupId, final boolean displayed, final List<MenuProductRequest> menuProducts) {
        this.name = name;
        this.price = price;
        this.menuGroupId = menuGroupId;
        this.displayed = displayed;
        this.menuProducts = Collections.unmodifiableList(menuProducts);
    }

    public String getName() {
        return name;
    }

    public BigDecimal getPrice() {
        return price;
    }

    public UUID getMenuGroupId() {
        return menuGroupId;
    }

    public boolean isDisplayed() {
        return displayed;
    }

    public List<MenuProductRequest> getMenuProducts() {
        return new ArrayList<>(menuProducts);
    }
}

File: src/main/java/kitchenpos/menus/tobe/application/dto/CreateMenuRequest.kt
package kitchenpos.menus.tobe.application.dto

import java.util.UUID

data class CreateMenuRequest(
    val name: String,
    val price: Int,
    val groupId: UUID,
    val displayed: Boolean,
    val menuProducts: List<MenuProductRequest>,
)

data class MenuProductRequest(
    val quantity: Int,
    val price: Int,
    val productId: UUID,
)
